tfhe/benches/integer/oprf.rs

```diff
@@ -1,33 +1,75 @@
+use crate::utilities::{write_to_json, OperatorType};
 use concrete_csprng::seeders::Seed;
-use criterion::{black_box, criterion_group, Criterion};
+use criterion::{black_box, Criterion};
+use itertools::iproduct;
+use std::vec::IntoIter;
 use tfhe::integer::keycache::KEY_CACHE;
 use tfhe::integer::IntegerKeyKind;
 use tfhe::keycache::NamedParam;
 use tfhe::shortint::parameters::*;
 
-fn oprf(c: &mut Criterion) {
-    let bench_name = "integer_oprf";
+/// An iterator that yields a succession of combinations
+/// of parameters and a num_block to achieve a certain bit_size ciphertext
+/// in radix decomposition
+struct ParamsAndNumBlocksIter {
+    params_and_bit_sizes: itertools::Product<IntoIter<PBSParameters>, IntoIter<u64>>,
+}
 
-    let mut bench_group = c.benchmark_group(bench_name);
+impl Default for ParamsAndNumBlocksIter {
+    fn default() -> Self {
+        let params = vec![PARAM_MESSAGE_2_CARRY_2_KS_PBS.into()];
+        let bit_sizes = vec![64];
+        let params_and_bit_sizes = iproduct!(params, bit_sizes);
+        Self {
+            params_and_bit_sizes,
+        }
+    }
+}
 
-    let param = PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+impl Iterator for ParamsAndNumBlocksIter {
+    type Item = (PBSParameters, u64, u64);
 
-    let (_, sk) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+    fn next(&mut self) -> Option<Self::Item> {
+        let (param, bit_size) = self.params_and_bit_sizes.next()?;
+        let num_block =
+            (bit_size as f64 / (param.message_modulus().0 as f64).log(2.0)).ceil() as u64;
 
-    bench_group.bench_function(&format!("64-bits-oprf::{}", param.name()), |b| {
-        b.iter(|| {
-            _ = black_box(sk.par_generate_oblivious_pseudo_random_unsigned_integer(
-                Seed(0),
-                64,
-                32,
-            ));
-        })
-    });
+        Some((param, num_block, bit_size))
+    }
 }
 
-criterion_group!(oprf2, oprf);
+pub fn unsigned_oprf(c: &mut Criterion) {
+    let bench_name = "integer::unsigned_oprf";
+
+    let mut bench_group = c.benchmark_group(bench_name);
```

Comment:
any reason to keep it in a separate module ?

Reply:
The file `integer/bench.rs` is already quite big so I wanted to keep OPRF benchmarks away. 
This is the only reason.

Reply:
Does this work ? I see the utilities mod done a bit differently ?

Reply:
I don’t know if the params/number of bits correspond to what we do in the other file ?